http/api/transport.go

```diff
@@ -29,6 +31,9 @@ const protocol = "http"
 var (
 	errMalformedData = errors.New("malformed request data")
 	auth             mainflux.ThingsServiceClient
+	// regexp accept subtopic with name without '.' to avoid problems
+	//	with nats, / par of subtopic will be replaced with '.'
+	channelPartRegExp = regexp.MustCompile(`^/channels/([\w\-]+)/messages(/[^?\.]+)*$`)
```

Comment:
Maybe it's safer to allow only subtopics with alphanumeric characters, underscore, dash (`([\w\-])+`) and wildcard (`#`,  `+`, `$` or `*`). It will probably work, but I just don't like an idea of having any special character (e.g. `~`, `!` or  `%`) in subtopics.

Reply:
@dusanb94 @beres for this MQTT standard must be consulted. If it permits special chars in topics names then we must respect this.

Reply:
Hi dubsand, yes it's safer, but why accept wildcard chars? http/api has only Publish method, what did you think?

Reply:
Hi, @beres. I wrote this having MQTT and an idea of making subtopics as generic as possible on my mind. However, wildcards are related to subscriptions, so it should not be a problem here.